truncate/table_schema_test.go

```diff
@@ -0,0 +1,225 @@
+	// More detailed tests for target and exclude filters follow in subtests.
+	t.Run("filterTableSchemas", func(t *testing.T) {
+		for _, test := range []struct {
+			desc          string
+			schemas       []*tableSchema
+			targetTables  []string
+			excludeTables []string
+			want          []*tableSchema
+			wantErr       bool
+		}{
+			{
+				desc:          "Both target and exclude tables are empty.",
+				schemas:       []*tableSchema{t1, t2, t3},
+				targetTables:  nil,
+				excludeTables: nil,
+				want:          []*tableSchema{t1, t2, t3},
+				wantErr:       false,
+			},
+			{
+				desc:          "Target tables are specified.",
+				schemas:       []*tableSchema{t1, t2, t3},
+				targetTables:  []string{t1.tableName, t2.tableName},
+				excludeTables: nil,
+				want:          []*tableSchema{t1, t2},
+				wantErr:       false,
+			},
+			{
+				desc:          "Exclude tables are specified.",
+				schemas:       []*tableSchema{t1, t2, t3},
+				targetTables:  nil,
+				excludeTables: []string{t1.tableName, t2.tableName},
+				want:          []*tableSchema{t3},
+				wantErr:       false,
+			},
+			{
+				desc:          "Both target and exclude tables are specified.",
+				schemas:       []*tableSchema{t1, t2, t3},
+				targetTables:  []string{t1.tableName},
+				excludeTables: []string{t2.tableName},
+				want:          nil,
+				wantErr:       true,
+			},
+		} {
+			t.Run(test.desc, func(t *testing.T) {
+				got, err := filterTableSchemas(test.schemas, test.targetTables, test.excludeTables)
+				if test.wantErr {
+					if err == nil {
+						t.Errorf("test wants error, but no error returned")
+					}
+					return
+				}
+
+				if len(got) != len(test.want) {
+					t.Errorf("len(got) %d, len(want) %d", len(got), len(test.want))
+				}
+
+				if diff := cmp.Diff(got, test.want, filterTableSchemasOpts...); diff != "" {
+					t.Errorf("mismatch (-got +want):\n%s", diff)
+				}
+			})
+		}
+	})
+
+	// Detailed tests for targetFilterTableSchemas
+	t.Run("targetFilterTableSchemas", func(t *testing.T) {
+		for _, test := range []struct {
+			desc         string
+			schemas      []*tableSchema
+			targetTables []string
+			want         []*tableSchema
+		}{
+			{
+				desc:         "Include multiple tables",
+				schemas:      []*tableSchema{t1, t2, t3},
+				targetTables: []string{t1.tableName, t2.tableName},
+				want:         []*tableSchema{t1, t2},
+			},
+			{
+				desc:         "Do nothing when no target tables are passed.",
+				schemas:      []*tableSchema{t1, t2, t3},
+				targetTables: nil,
+				want:         []*tableSchema{t1, t2, t3},
+			},
+			// TODO: Determine the specifications for parent-child relationships in hierarchical interleaved tables, and add corresponding tests and implementation.
+			// This includes defining the behavior of targetFilterTableSchemas for cases where tables not included in the target list are subject to cascade deletion.
+		} {
```

Comment:
Currently, when a parent table is specified as a target table, interleaved child tables are not printed. However, I am considering changing the specification to print these interleaved child tables as well. What are your thoughts on this?

Reply:
Sounds good. If I understand correctly, the final outcome won't be changed (i.e. interleaved tables are also deleted) and it just changes how the target tables are prompted to users?

If so, maybe different PR might be better 👍

Reply:
Your understanding is correct.
I've opened a new issue regarding this problem (#18).

I noticed that Spanner already supports cascade deletion for foreign keys (see [Release Notes, July 18, 2023](https://cloud.google.com/spanner/docs/release-notes?hl=ja#July_18_2023)), so I've opened another issue (#19).

In this PR, we focus on the #8 for cascade deletion on interleaved tables.